Session/Calls/CallVC.swift

```diff
@@ -734,15 +726,55 @@ final class CallVC: UIViewController, VideoPreviewDelegate, AVRoutePickerViewDel
             // Added delay of preview due to permission dialog alert dismissal on allow.
             // It causes issue on `VideoPreviewVC` presentation animation,
             // If camera permission is already allowed no animation delay is needed
-            var previewDelay = Permissions.camera == .undetermined ? 0.5 : 0
+            let previewDelay = Permissions.camera == .undetermined ? 0.5 : 0
             
-            Permissions.requestCameraPermissionIfNeeded(presentingViewController: self, using: dependencies) { [weak self] isAuthorized in
-                guard isAuthorized else { return }
+            Permissions.requestCameraPermissionIfNeeded(
+                useCustomDeniedAlert: true,
+                using: dependencies
+            ) { [weak self, dependencies] isAuthorized in
                 
-                DispatchQueue.main.asyncAfter(deadline: .now() + previewDelay) { [weak self] in
-                    let previewVC = VideoPreviewVC()
-                    previewVC.delegate = self
-                    self?.present(previewVC, animated: true, completion: nil)
+                let status = Permissions.camera
+
+                switch (isAuthorized, status) {
+                    case (false, .denied):
+                        guard let presentingViewController: UIViewController = (self?.navigationController ?? dependencies[singleton: .appContext].frontMostViewController)
+                        else { return }
+
+                        DispatchQueue.main.async {
+                            let confirmationModal: ConfirmationModal = ConfirmationModal(
+                                info: ConfirmationModal.Info(
+                                    title: "cameraAccessRequired".localized(),
+                                    body: .attributedText(
+                                        "cameraAccessDeniedMessage"
+                                            .put(key: "app_name", value: Constants.app_name)
+                                            .localizedFormatted(),
+                                        scrollMode: .never
+                                    ),
+                                    confirmTitle: "endCallToEnable".localized(),
+                                    confirmStyle: .danger,
+                                    cancelTitle: "Remind Me Later", // stringlint:ignore
```

Comment:
This needs to be localised - can you please ask @pottsandpans to add it to Crowdin? (we generally should never hardcode user-facing strings)